expan/core/early_stopping.py

```diff
@@ -316,40 +292,107 @@ def f(x, y):
 
-    credibleMass = 0.95                # another magic number
-    leftOut      = 1.0 - credibleMass
-    p1           = round(leftOut/2.0, 5)
-    p2           = round(1.0 - leftOut/2.0, 5)
+    credible_mass = 0.95
+    left_out      = 1.0 - credible_mass
+    p1            = round(left_out/2.0, 5)
+    p2            = round(1.0 - left_out/2.0, 5)
 
-    credible_interval_delta            = HDI_from_MCMC(trace_absolute_effect_size, credibleMass)
-    credible_interval_delta_normalized = HDI_from_MCMC(trace_normalized_effect_size, credibleMass)
+    credible_interval_delta            = HDI_from_MCMC(trace_absolute_effect_size, credible_mass)
+    credible_interval_delta_normalized = HDI_from_MCMC(trace_normalized_effect_size, credible_mass)
 
     stop = credible_interval_delta_normalized[1] - credible_interval_delta_normalized[0] < posterior_width
 
-    return {'stop'                  : bool(stop),
-            'delta'                 : float(mu_x - mu_y),
-            'confidence_interval'   : [{'percentile': p*100, 'value': v} for p, v in zip([p1, p2], credible_interval_delta)],
-            'treatment_sample_size' : int(n_x),
-            'control_sample_size'   : int(n_y),
-            'treatment_mean'        : float(mu_x),
-            'control_mean'          : float(mu_y),
-            'number_of_iterations'  : num_iters}
+    treatment_statistics = SampleStatistics(int(n_x), float(mu_x), float(np.nanvar(x)))
+    control_statistics   = SampleStatistics(int(n_y), float(mu_y), float(np.nanvar(y)))
+    variant_statistics   = BaseTestStatistics(control_statistics, treatment_statistics)
+
+    return EarlyStoppingTestStatistics(variant_statistics.control_statistics,
+                                       variant_statistics.treatment_statistics,
+                                       float(mu_x - mu_y),
+                                       dict([(p * 100, v) for p, v in zip([p1, p2], credible_interval_delta)]),
+                                       None, None, stop)
+
+
+def get_trace_normalized_effect_size(distribution, traces):
+    """ Obtaining a Stan model statistics for 'normal' or 'poisson' distribution
+
+    :param distribution: name of the KPI distribution model, which assumes a Stan model file with the same name exists
+    :type  distribution: str
+    :param traces: sampling statistics
+    :type  traces: dict
+
+    :return: sample of data points from posterior distribution of some parameter
+    :rtype:  array-like
+    """
+    # check of traces type, if traces is not dict "traces['alpha']" will return KeyError
+    if not isinstance(traces, dict):
+        raise TypeError("Traces statistics is not a dictionary and does not contain alpha or delta statistics.")
+
+    if distribution == 'normal':
+        return traces['alpha']
+    elif distribution == 'poisson':
+        variance = np.nanmean(np.array(traces['delta']))
+        return traces['delta'] / np.sqrt(np.absolute(variance))
```

Comment:
Just a remark for the future work (not part of this pull request):

1. for a normal distribution, the returned effect size is already normalized
2. for a Poisson distribution, we have to normalize it by hand

Maybe we should stick to one particular scheme and return a non-normalized effect size from the normal model (or the normalized from the Poisson model).

Reply:
good point! I've created a issue #200 to keep track of it. :)